### DISCUSSION TITLE
Ever since updating to v00.04-beta, I get "probing/leveling" process on each print. Can I disable that?

### DISCUSSION THREAD

**Discussion by DougJoseph** [xblax/flashforge_ad5m_klipper_mod#106] [Q&A] [answered]
Ever since updating to v00.04-beta, I get "probing/leveling" process on each print. Can I disable that? 

I redid the resonance vibration test and the probing/leveling process, and saved and rebooted, yet I still get a redo of the probing/leveling process on every print.

**Comment by Tiwatz** [comment]
>Can I disable that?
> 

If there is no bed mesh profile loaded, the START_PRINT macro will do a new one. If you want to load your previous mesh you can add:
`BED_MESH_PROFILE LOAD=auto`
Before the START_PRINT line in your slicer.

**Reply by xblax** [reply]
Not that's not really needed. If an automatic profile should be loaded on Klipper start you can just save the 'default' profile. But I would not recommend it unless always printing with the same bed temp.

**Reply by DougJoseph** [reply]
I am pretty much always printing with the same bed temp. On the v00.03, the saving of the bed mesh seemed to work and stay in memory. On the v00.04 I seem to be struggling to get the bed mesh to "stay saved." More soon.

**Reply by xblax** [reply]
Yes sure that's ok, You can just rename the profile that was created automatically to "default" and save it.  The "default" profile is automatically enabled by Klipper. Or manually load any other profile before printing. Then no automatic leveling will be done.

The mod does by design not try to guess how you want to handle the profiles. The only default logic is, that if no profile is loaded before printing it will automatically create one, because printing without profile is not possible.

**Reply by Tiwatz** [reply]
Are you sure that the "default" profile is automatically enabled? I know it used to be but in later versions of Klipper it's not. I'm not sure when they disabled that.

**Reply by DougJoseph** [reply]
OK, makes sense. I had already done the SAVE_CONFIG command, so I felt like it was supposed to have already been saved. I just now googled "klipper how to save bed mesh as default" and the first result said, 

> Save the Mesh
> Always use the SAVE_CONFIG command to store the mesh and restart Klipper. The values will be saved at the bottom of the config file, under the SAVE_CONFIG section. You should not be editing that part. 

So I checked my printer.cfg file, and it dutifully shows:

<img width="1728" alt="Screenshot 2024-03-26 at 3 32 06 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/825a0850-6838-41a7-aa73-db57225997ca">

So, I *think* my mesh is saved "as the default" but maybe there is something else that needs done? Maybe after all I do need to do what [Tiwatz](https://github.com/Tiwatz) suggested, 

`BED_MESH_PROFILE LOAD=auto`

**Reply by xblax** [reply, marked as answer]
> Are you sure that the "default" profile is automatically enabled? I know it used to be but in later versions of Klipper it's not. I'm not sure when they disabled that.

Yes in Klipper 0.11 it's still in: https://github.com/Klipper3d/klipper/commit/4df1adeb2e168b9b87856e738d3109a57ba1cac6

@DougJoseph
No, you saved the bed mesh named "auto". You have to rename it and then save. It is purposely named "auto" to not clash with a user-defined default and not automatically load if it's saved. If you want to make it default, that must be done explicitly. Check the bed mesh tab in Mainsail or Fluidd, there is a rename function.

You can also have multiple meshes saved, i.e. I have "pla" and "petg" saved. If one is loaded before printing, no "auto" mesh is created.

**Reply by DougJoseph** [reply]
Ah, got it! This is good to know! Glad to learn this. Thank you!

**Reply by DougJoseph** [reply]
Hmm. Well, I went to rename it in Mainsail, and it would not let me save it as "default." I got this warning message:

Profile 'default' is reserved, please choose another profile name.

<img width="432" alt="Screenshot 2024-03-26 at 4 20 23 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/c7b647d3-5136-4ed7-975e-5f86491dfddd">

**Reply by DougJoseph** [reply]
I currently have my Start GCode including this:

`BED_MESH_PROFILE LOAD=auto`

so maybe I'm good now.

**Reply by xblax** [reply]
Ugh, that's unfortunate. Didn't know that. Have to check it later, my printer is busy printing at the moment.
I think you can just edit the profile name in printer.cfg in the mean time (even though it says to not edit it).

**Reply by DougJoseph** [reply]
Thanks! :-)

**Reply by DougJoseph** [reply]
Just circling back to confirm that after I manually renamed the mesh to "default" using the config editor in Mainsail, now my prints don't always do a mesh probing on every print! Thanks!

**Comment by DougJoseph** [comment]
I gather that there is a way to set  up PLA profile that uses PLA mesh, and PETG profile that uses PETG mesh. I don't know how to do that, but I'm guessing it's done somewhere in the filament choice/settings in OrcaSlicer.

**Reply by consp** [reply]
I have two fillament profiles and mesh levels, names pla and abs which auto-load as fillament gcode (every slic3r based slicer can do that), added `BED_MESH_PROFILE LOAD=name` to both the profiles so it auto-switches before starting print.

You could do the same for different nozzels but I would remesh them after switching anyway

**Reply by DougJoseph** [reply]
Will give this a try. Thanks!
